src/silx/gui/plot/ImageStack.py

```diff
@@ -197,23 +240,27 @@ class _ToggleableUrlSelectionTable(qt.QWidget):
     sigCurrentUrlChanged = qt.Signal(str)
     """Signal emitted when the active/current url change"""
 
+    sigUrlsRemoved = qt.Signal(tuple)
+
     def __init__(self, parent=None) -> None:
-        qt.QWidget.__init__(self, parent)
+        super().__init__(parent)
         self.setLayout(qt.QGridLayout())
         self._toggleButton = qt.QPushButton(parent=self)
         self.layout().addWidget(self._toggleButton, 0, 2, 1, 1)
         self._toggleButton.setSizePolicy(qt.QSizePolicy.Fixed,
                                          qt.QSizePolicy.Fixed)
 
         self._urlsTable = UrlList(parent=self)
+
         self.layout().addWidget(self._urlsTable, 1, 1, 1, 2)
 
         # set up
         self._setButtonIcon(show=True)
 
         # Signal / slot connection
         self._toggleButton.clicked.connect(self.toggleUrlSelectionTable)
-        self._urlsTable.sigCurrentUrlChanged.connect(self._propagateSignal)
+        self._urlsTable.sigCurrentUrlChanged.connect(self._propagateCurrentUrlChangedSignal)
+        self._urlsTable.sigUrlsRemoved.connect(self._propageUrlsRemovedSignal)
```

Comment:
I think you can connect signals directly and spare the `_propage*` methods:

```suggestion
        self._urlsTable.sigCurrentUrlChanged.connect(self.sigCurrentUrlChanged)
        self._urlsTable.sigUrlsRemoved.connect(self.sigUrlsRemoved)
```

Reply:
Nice, I was aware that you could connect directly a signal